source/common/ssl/context_config_impl.cc

```diff
@@ -1,17 +1,40 @@
 #include "common/ssl/context_config_impl.h"
 
+#include <memory>
 #include <string>
 
 #include "common/common/assert.h"
 #include "common/config/datasource.h"
 #include "common/config/tls_context_json.h"
 #include "common/protobuf/utility.h"
+#include "common/ssl/tls_certificate_config_impl.h"
 
 #include "openssl/ssl.h"
 
 namespace Envoy {
 namespace Ssl {
 
+namespace {
+
+template <typename T>
+const std::shared_ptr<T> findSecretAndCheckType(const Secret::SecretManager& secret_manager,
```

Comment:
This function will end up getting used for other types of secrets. I would add this to the primary interface.

Reply:
Moved to SecretManager interface